File: javascript/darts/darts.js
export const solve = (x, y) => {
  if (typeof x !== 'number' || typeof y !== 'number') return null;
  const d = Math.sqrt(x ** 2 + y ** 2);
  if (d > 10) return 0;
  if (d > 5) return 1;
  if (d > 1) return 5;
  return 10;
};

File: kotlin/darts/src/main/kotlin/Darts.kt
import kotlin.math.*

object Darts {

    fun score(x: Number, y: Number): Int {
        when(hypot(x.toDouble(), y.toDouble())) {
            in 0.0..1.0 -> return 10
            in 1.0..5.0 -> return 5
            in 5.0..10.0 -> return 1
            else -> return 0
        }
    }

}
